stablehlo/reference/Element.h

```diff
@@ -52,7 +57,7 @@ class Element {
 
  private:
   Type type_;
-  Attribute value_;
+  std::variant<APInt, std::vector<APFloat>> value_;
```

Comment:
We should use a variant of: `APInt`, `APFloat` and `std::complex<APFloat>` - to explicitly specify how we're modelling complex numbers.

Reply:
Tried using the suggested one  `std::variant<APInt, APFloat, std::complex<APFloat>>` but it gives errors related to calling the [private constructor](https://github.com/llvm-mirror/llvm/blob/2c4ca6832fa6b306ee6a7010bfb80a3f2596f824/include/llvm/ADT/APFloat.h#L832) of APFloat. As per the [requirement](https://en.cppreference.com/w/cpp/utility/variant) of std::variant, the first alternative of the variant  do have a default constructor, so not sure what exactly the problem was. I will explore it later. 

Hence using the following instead.

```
std::variant<APInt, APFloat, std::pair<APFloat, APFloat>>
```

Reply:
As promised, here's a pull request that compiles on my machine. Can you take a look if it compiles on yours? https://github.com/sdasgup3/stablehlo/pull/2.

Reply:
Let us take this up in a [new ticket](https://github.com/openxla/stablehlo/issues/226)